woocommerce/woocommerce-gutenberg-products-block
# Fix an issue in which sale price didn't have a strikethrough on mobile and correct shade on desktop.

Fixes #2991
Fixes #3080 (edit - @haszari)



This fixes two issues with sale price:

- It was the same color as `normal price` on desktop (it should have been a lesser shade).
![image](https://user-images.githubusercontent.com/6165348/91748531-047c8280-ebb8-11ea-833a-ed6d112ad8aa.png)
This issue did not exist on mobile, because `del` and its display value.
- It didn't have a strikethrough on mobile, but it did have on desktop, strikethrough is provided by `del`
![image](https://user-images.githubusercontent.com/6165348/91748657-3beb2f00-ebb8-11ea-9957-21e539655a9a.png)

Both issues stem from `<del>` and what you assign for a display value for its children.

WooCommerce core provides `opacity: 0.5` for `.sale del`, this works if `<del>` children's display value is an inline value (inline, inline-block, inline-table, inline-grid, inline-flex), but not `block` or any block level display value.
`<del>` natively styles inner text with `strikethrough` but that will not work if children's display value is either `inline-table` or `inline-block`, but all other values, block or inline, will work.
Clearly this is too much logic to juggle, so I reached for `del` and any accessibility issues we might face if we delete it, turns out it provides no value whatsoever over `text-decoration` both will not be announced, sources:
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/del
- http://pauljadam.com/demos/css-line-through-del-ins-accessibility.html

So I deleted the tag.




### Screenshots
![image](https://user-images.githubusercontent.com/6165348/91749604-a18beb00-ebb9-11ea-83cf-d1cf589e973c.png)
![image](https://user-images.githubusercontent.com/6165348/91749617-a6509f00-ebb9-11ea-9e78-77a4cf079925.png)
![image](https://user-images.githubusercontent.com/6165348/91749648-b23c6100-ebb9-11ea-91bf-c3018f07c9c3.png)



### How to test the changes in this Pull Request:

1. Add a product with sale to your cart.
2. Check the product, on both mobile and desktop, the sale price should be gray and has a strikethrough.
3. Check All Product, the same should be applied there.
